coalib/io/Directory.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+from coalib.parsing.Globbing import relative_recursive_glob
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('path', 'timestamp')
+class Directory:
+    """
+    The `Directory` class acts as an interface to directories and
+    provides useful information about them such as:
+
+        * The directory path.
+        * The parent directory path.
+        * The last modified timestamp of the directory.
+        * The number of children contained inside a directory.
```

Comment:
Not just the number, the children itself ;)
--> "The children contained inside ~~a~~ *the* directory"